logr.go

```diff
@@ -250,6 +287,44 @@ func (l Logger) WithSink(sink LogSink) Logger {
 type Logger struct {
 	sink  LogSink
 	level int
+
+	// This is a pointer to keep Logger as small as possible.
+	// The effect of having to allocate contextKV instances
+	// gets mitigated via sync.Pool.
+	contextKV *contextKV
+}
+
+// contextKV is shared by different Logger instances. It combines everything
+// that is needed to retrieve key/value pairs from a context.
+type contextKV struct {
```

Comment:
naming: how about `contextHandle` or `contextMeta` or something - KV makes it sound like a single value.

Reply:
I switched to `contextMeta` for the the type and to `context` for the `Logger` field.